src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -0,0 +1,167 @@
+  private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+  // TODO: Get the proper scopes. CalendarScopes might be outdated and not have all the proper scopes.
+  private static final List<String> SCOPES = Lists.newArrayList(CalendarScopes.all());
+
+  /**
+   * Obtain a list of events from the specified group if given the Calendar ID.
+   * Obtain the Calendar ID of a group given the Group ID. If both provided, return list of events.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String calendarId = getParameter(request, "calendarid", "");
+    String groupId = getParameter(request, "groupid", "");
+
+    if (!Strings.isNullOrEmpty(calendarId)) {
+      try {
+        Calendar service = getCalendarService(); 
+
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(getEventsList(service, calendarId)));
+      } catch (Exception authenticationError) {
+        authenticationError.printStackTrace();
+        response.getWriter().println("Invalid credentials or calendar ID");
+      }
+    } else if (!Strings.isNullOrEmpty(groupId)) {
+      try {
+        calendarId = getCalendarId(groupId);
+
+        response.setContentType("text/plain");
+        response.getWriter().println(calendarId);
+      } catch (Exception entityError) {
+        entityError.printStackTrace();
+        response.getWriter().println("Invalid group ID");
+      }
+    }
+  }
+
+  /**
+   * Create an event and add it to the calendar of the specified group
+   * POST body should contain: title, location (opt), description (opt), start time, and end time.
+   */
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String groupId = getParameter(request, "groupid", "");
+    String eventTitle = getParameter(request, "title", "");
+    String eventStart = getParameter(request, "start", "");
+    String eventEnd = getParameter(request, "end", "");
+
+    if(!Strings.isNullOrEmpty(groupId) && !Strings.isNullOrEmpty(eventTitle) 
+        && !Strings.isNullOrEmpty(eventStart) && !Strings.isNullOrEmpty(eventEnd)) {
+      try {
+        Event event = addEvent(getCalendarService(), getCalendarId(groupId), 
+            eventTitle, getParameter(request, "location", ""),
+            getParameter(request, "description", ""), eventStart, eventEnd);
+
+        response.setContentType("text/plain");
+        response.getWriter().println(event.getHtmlLink());
+      } catch (Exception entityError) {
+        entityError.printStackTrace();
+        response.getWriter().println("Invalid input");
+      }
+    }
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
if you're not planning on having any other default value other than an empty string, you could remove that third parameter `defaultValue` and just have it be `if (value == null) { return ""; }`